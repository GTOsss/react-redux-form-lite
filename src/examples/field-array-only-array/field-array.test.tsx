import ReduxThunkTester from 'redux-thunk-tester';
import {applyMiddleware, combineReducers, createStore} from 'redux';
import {reducer} from '../../index';
import {mount} from 'enzyme';
import {Provider} from 'react-redux';
import getFieldArray from './field-array';
import * as React from 'react';

const renderComponent = (pushUser?) => {
  const reduxThunkTester = new ReduxThunkTester();
  const FieldArray = getFieldArray();

  const store = createStore(
    combineReducers({reduxForm: reducer}),
    applyMiddleware(reduxThunkTester.createReduxThunkHistoryMiddleware()),
  );

  const component = mount(
    <Provider store={store}>
      <FieldArray pushUser={pushUser} />
    </Provider>,
  );

  return {
    reduxThunkTester,
    getFormState: () => store.getState().reduxForm.fieldArrayExample as IReduxFormState<any>,
    component,
  };
};

describe('<FieldArrayOnlyArray />', () => {
  test('Store field-array after 1 push', () => {
    const {component, getFormState} = renderComponent();
    component.find('#addUser').simulate('click');
    expect(getFormState()).toMatchSnapshot();
  });

  test('Store field-array after 1 push, remove 1', () => {
    const {component, getFormState} = renderComponent();
    component.find('#addUser').simulate('click');
    component.find('#removeUser').at(0).simulate('click');
    expect(getFormState()).toMatchSnapshot();
  });

  test('Store field-array after 3 push, remove 2 (remove middle)', () => {
    const {component, getFormState} = renderComponent();
    component.find('#addUser').simulate('click');
    component.find('input').at(0).simulate('change', {target: {value: '1'}});
    component.find('input').at(1).simulate('change', {target: {value: '1'}});
    component.find('input').at(2).simulate('change', {target: {value: '1'}});
    component.find('#addUser').simulate('click');
    component.find('input').at(3).simulate('change', {target: {value: '2'}});
    component.find('input').at(4).simulate('change', {target: {value: '2'}});
    component.find('input').at(5).simulate('change', {target: {value: '2'}});
    component.find('#addUser').simulate('click');
    component.find('input').at(6).simulate('change', {target: {value: '3'}});
    component.find('input').at(7).simulate('change', {target: {value: '3'}});
    component.find('input').at(8).simulate('change', {target: {value: '3'}});
    component.find('#removeUser').at(1).simulate('click');
    expect(getFormState()).toMatchSnapshot();
  });

  test('Store field-array after 4 push, remove 2 (remove middle), submit', () => {
    const {component, getFormState} = renderComponent();
    component.find('#addUser').simulate('click');
    component.find('input').at(0).simulate('change', {target: {value: '1'}});
    component.find('input').at(1).simulate('change', {target: {value: '1'}});
    component.find('input').at(2).simulate('change', {target: {value: '1'}});
    component.find('#addUser').simulate('click');
    component.find('input').at(3).simulate('change', {target: {value: '2'}});
    component.find('input').at(4).simulate('change', {target: {value: '2'}});
    component.find('input').at(5).simulate('change', {target: {value: '2'}});
    component.find('#addUser').simulate('click');
    component.find('input').at(6).simulate('change', {target: {value: '3'}});
    component.find('input').at(7).simulate('change', {target: {value: '3'}});
    component.find('input').at(8).simulate('change', {target: {value: '3'}});
    component.find('#addUser').simulate('click');
    component.find('input').at(9).simulate('change', {target: {value: '4'}});
    component.find('input').at(10).simulate('change', {target: {value: '4'}});
    component.find('input').at(11).simulate('change', {target: {value: '4'}});
    component.find('#removeUser').at(2).simulate('click');
    component.find('form').simulate('submit');
    expect(getFormState()).toMatchSnapshot();
  });
});
