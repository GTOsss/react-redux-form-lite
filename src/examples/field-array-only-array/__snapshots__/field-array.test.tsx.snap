// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FieldArrayOnlyArray /> Store field-array after 1 push 1`] = `
Object {
  "form": Object {
    "activeField": "",
    "blurred": false,
    "changed": false,
    "errorsMap": Object {},
    "focused": false,
    "hasErrors": false,
    "hasWarnings": false,
    "submitted": false,
    "warningsMap": Object {},
  },
  "meta": Object {
    "users": Array [
      Object {
        "about": Object {
          "hobby": Object {
            "active": false,
            "blurred": false,
            "changed": false,
            "error": "",
            "focused": false,
            "warning": "",
          },
        },
        "firstName": Object {
          "active": false,
          "blurred": false,
          "changed": false,
          "error": "",
          "focused": false,
          "warning": "",
        },
        "secondName": Object {
          "active": false,
          "blurred": false,
          "changed": false,
          "error": "",
          "focused": false,
          "warning": "",
        },
      },
    ],
  },
  "values": Object {
    "users": Array [
      Object {
        "about": Object {
          "hobby": undefined,
        },
        "firstName": undefined,
        "friends": Array [],
        "id": 0,
        "secondName": undefined,
      },
    ],
  },
}
`;

exports[`<FieldArrayOnlyArray /> Store field-array after 1 push, remove 1 1`] = `
Object {
  "form": Object {
    "activeField": "",
    "blurred": false,
    "changed": false,
    "errorsMap": Object {},
    "focused": false,
    "hasErrors": false,
    "hasWarnings": false,
    "submitted": false,
    "warningsMap": Object {},
  },
  "meta": Object {
    "users": Array [],
  },
  "values": Object {
    "users": Array [],
  },
}
`;

exports[`<FieldArrayOnlyArray /> Store field-array after 3 push, remove 2 (remove middle) 1`] = `
Object {
  "form": Object {
    "activeField": "",
    "blurred": false,
    "changed": true,
    "errorsMap": Object {},
    "focused": false,
    "hasErrors": false,
    "hasWarnings": false,
    "submitted": false,
    "warningsMap": Object {},
  },
  "meta": Object {
    "users": Array [
      Object {
        "about": Object {
          "hobby": Object {
            "active": false,
            "blurred": false,
            "changed": true,
            "error": "",
            "focused": false,
            "warning": "",
          },
        },
        "firstName": Object {
          "active": false,
          "blurred": false,
          "changed": true,
          "error": "",
          "focused": false,
          "warning": "",
        },
        "secondName": Object {
          "active": false,
          "blurred": false,
          "changed": true,
          "error": "",
          "focused": false,
          "warning": "",
        },
      },
      Object {
        "about": Object {
          "hobby": Object {
            "active": false,
            "blurred": false,
            "changed": true,
            "error": "",
            "focused": false,
            "warning": "",
          },
        },
        "firstName": Object {
          "active": false,
          "blurred": false,
          "changed": true,
          "error": "Min length should be 10",
          "focused": false,
          "warning": "",
        },
        "secondName": Object {
          "active": false,
          "blurred": false,
          "changed": true,
          "error": "",
          "focused": false,
          "warning": "",
        },
      },
    ],
  },
  "values": Object {
    "users": Array [
      Object {
        "about": Object {
          "hobby": "1",
        },
        "firstName": "1",
        "friends": Array [],
        "id": 0,
        "secondName": "1",
      },
      Object {
        "about": Object {
          "hobby": "3",
        },
        "firstName": "3",
        "friends": Array [],
        "id": 2,
        "secondName": "3",
      },
    ],
  },
}
`;

exports[`<FieldArrayOnlyArray /> Store field-array after 3 push, remove 2 (remove middle), submit 1`] = `
Object {
  "form": Object {
    "activeField": "",
    "blurred": false,
    "changed": true,
    "errorsMap": Object {},
    "focused": false,
    "hasErrors": false,
    "hasWarnings": false,
    "submitted": true,
    "warningsMap": Object {},
  },
  "meta": Object {
    "users": Array [
      Object {
        "about": Object {
          "hobby": Object {
            "active": false,
            "blurred": false,
            "changed": true,
            "error": "",
            "focused": false,
            "warning": "",
          },
        },
        "firstName": Object {
          "active": false,
          "blurred": false,
          "changed": true,
          "error": "",
          "focused": false,
          "warning": "",
        },
        "secondName": Object {
          "active": false,
          "blurred": false,
          "changed": true,
          "error": "",
          "focused": false,
          "warning": "",
        },
      },
      Object {
        "about": Object {
          "hobby": Object {
            "active": false,
            "blurred": false,
            "changed": true,
            "error": "",
            "focused": false,
            "warning": "",
          },
        },
        "firstName": Object {
          "active": false,
          "blurred": false,
          "changed": true,
          "error": "",
          "focused": false,
          "warning": "",
        },
        "secondName": Object {
          "active": false,
          "blurred": false,
          "changed": true,
          "error": "",
          "focused": false,
          "warning": "",
        },
      },
    ],
  },
  "values": Object {
    "users": Array [
      Object {
        "about": Object {
          "hobby": "1",
        },
        "firstName": "1",
        "friends": Array [],
        "id": 0,
        "secondName": "1",
      },
      Object {
        "about": Object {
          "hobby": "3",
        },
        "firstName": "3",
        "friends": Array [],
        "id": 2,
        "secondName": "3",
      },
    ],
  },
}
`;

exports[`<FieldArrayOnlyArray /> Store field-array after 4 push, remove 2 (remove middle), submit 1`] = `
Object {
  "form": Object {
    "activeField": "",
    "blurred": false,
    "changed": true,
    "errorsMap": Object {
      "users": Array [
        undefined,
        undefined,
        Object {
          "firstName": "Min length should be 10",
        },
      ],
    },
    "focused": false,
    "hasErrors": true,
    "hasWarnings": false,
    "submitted": true,
    "warningsMap": Object {},
  },
  "meta": Object {
    "users": Array [
      Object {
        "about": Object {
          "hobby": Object {
            "active": false,
            "blurred": false,
            "changed": true,
            "error": "",
            "focused": false,
            "warning": "",
          },
        },
        "firstName": Object {
          "active": false,
          "blurred": false,
          "changed": true,
          "error": "",
          "focused": false,
          "warning": "",
        },
        "secondName": Object {
          "active": false,
          "blurred": false,
          "changed": true,
          "error": "",
          "focused": false,
          "warning": "",
        },
      },
      Object {
        "about": Object {
          "hobby": Object {
            "active": false,
            "blurred": false,
            "changed": true,
            "error": "",
            "focused": false,
            "warning": "",
          },
        },
        "firstName": Object {
          "active": false,
          "blurred": false,
          "changed": true,
          "error": "",
          "focused": false,
          "warning": "",
        },
        "secondName": Object {
          "active": false,
          "blurred": false,
          "changed": true,
          "error": "",
          "focused": false,
          "warning": "",
        },
      },
      Object {
        "about": Object {
          "hobby": Object {
            "active": false,
            "blurred": false,
            "changed": true,
            "error": "",
            "focused": false,
            "warning": "",
          },
        },
        "firstName": Object {
          "active": false,
          "blurred": false,
          "changed": true,
          "error": "Min length should be 10",
          "focused": false,
          "warning": "",
        },
        "secondName": Object {
          "active": false,
          "blurred": false,
          "changed": true,
          "error": "",
          "focused": false,
          "warning": "",
        },
      },
    ],
  },
  "values": Object {
    "users": Array [
      Object {
        "about": Object {
          "hobby": "1",
        },
        "firstName": "1",
        "friends": Array [],
        "id": 0,
        "secondName": "1",
      },
      Object {
        "about": Object {
          "hobby": "2",
        },
        "firstName": "2",
        "friends": Array [],
        "id": 1,
        "secondName": "2",
      },
      Object {
        "about": Object {
          "hobby": "4",
        },
        "firstName": "4",
        "friends": Array [],
        "id": 3,
        "secondName": "4",
      },
    ],
  },
}
`;
